### ISSUE TITLE
MinMaxScaler -- not fitting entire min/max values?

### ISSUE THREAD

**Issue by mwahal** [DarkKnight1991/Stock-Price-Prediction#5]
In the code

```
# scale the feature MinMax, build array
x = df_train.loc[:,train_cols].values
min_max_scaler = MinMaxScaler()
x_train = min_max_scaler.fit_transform(x)
x_test = min_max_scaler.transform(df_test.loc[:,train_cols])
```

You are scaling the min/max from the train values and then fitting the test. What if the maximum value of X is higher in test vs train? How will it work in x_test after the transformation? The value will be more than 1 for the maximum X in test set which was higher than Xmas in train ? 

Thanks
Mudit

**Comment by nayash** [comment]
yes, that's what happens while min-max scaling. You can easily verify this for yourself like this:
Here MinMaxScaler "fits" on each column of x1 and transforms x2. As you can see the values less than minimum value of source (x1) becomes -ve (check (3,0) of scaled_x2). The idea is to not let the trained model see whole data.
![Screenshot from 2020-09-29 21-14-30](https://user-images.githubusercontent.com/25876491/94581740-0ad64b00-0299-11eb-8327-73cbb59a49c4.png)
